fetchai/agents-aea
# [AEA-482] Fix stub connection

### Changes
- Fix input file parsing: file was read as a whole and parsed as one line
  + otherwise produces and errors everytime there is more than one message in the file
- Fix truncating: after reading from input file, add seek call to move fd pointer to the start of the file
  + otherwise misses messages
- Fix output format: to allow another process to use the output file as its input file
  + needs cleaning
- Add file locking/unlocking:  avoids race conditions

### How to test
Examples at https://github.com/fetchai/agents-p2p#stub-connection

Note: for PR history check https://github.com/fetchai/agents-aea/pull/1060

### flaky tests 
- https://github.com/fetchai/agents-aea/pull/1076/checks?check_run_id=604898661#step:6:11337
